packages/react-components/src/shared/src/useControllableState.ts

```diff
@@ -42,7 +49,7 @@ function useComputeInitialState<T>(controlledValue: T, initialValue: T, defaultV
     return result(state, isControlled, true);
 }
 
-function computeSubsequentState<T>(controlledValue: T, currentState: T, isControlled: boolean) {
+function computeSubsequentState<T>(controlledValue: T, currentState: T, isControlled: boolean): { newState: T; hasChanged: boolean; } {
```

Comment:
Are we fine with this synthax?

Reply:
Looks good, what are the alternatives?

Reply:
naming the type using an interface

Reply:
Looks fined, finding a name here would be quite hard.